### PR TITLE
Improved Implementation of Plugins for 3.0

### PR DESCRIPTION
**Pull request by rjmatthews62**
This is part of an ongoing process to implement the plugin interface, which has radically changed in 3.0.

* Scripts: walk, random and colornote modified with required support to work in 3.0
* Wrappers built for needed classes
* set("property") implemented.
* readProperties and writeProperties implemented for use in metadata. (see: proplist.qml)
* ElementW handling improved to reduce memory leakage.

### CODE REVIEW

**Comment by lasconic** [comment]
Also, you probably want to have a look to the tests in mtest/scripting.

**Comment by Jojo-Schmitz** [comment]
Esp. at https://travis-ci.org/musescore/MuseScore/jobs/279393360#L4897-L4916

File: libmscore/accidental.cpp [hunk 1]
```diff
@@ -439,7 +439,13 @@ bool Accidental::setProperty(P_ID propertyId, const QVariant& v)
...
       triggerLayout();
       return true;
-      }
+}
+
+void Accidental::supportedProperties(QList<P_ID>& dest, bool writeable)
+{
+  Element::supportedProperties(dest, writeable);
+  dest << P_ID::SMALL << P_ID::ACCIDENTAL_BRACKET << P_ID::ROLE;
+}
```
Comment (Jojo-Schmitz) [inline libmscore/accidental.cpp]: Wrong indentation for the entire hunk

File: libmscore/chord.cpp [hunk 2]
```diff
@@ -3400,6 +3400,29 @@ Shape Chord::shape() const
...
+      for (Chord* c : chord()->_graceNotes) {
+            _gnotes << dynamic_cast<ChordW*>(ElementW::buildWrapper(c));
+            }
+      return QmlListAccess<ChordW>(this, _gnotes);
+      }
+
+QQmlListProperty<NoteW> ChordW::qmlNotes()
+      {
+      _notes.clear();
+      for(Note* n : chord()->notes()) {
```
Comment (Jojo-Schmitz) [inline libmscore/chord.cpp]: space between `for` and `(`

File: libmscore/chordrest.cpp [hunk 3]
```diff
@@ -1107,7 +1107,17 @@ bool ChordRest::setProperty(P_ID propertyId, const QVariant& v)
...
+
+void ChordRest::supportedProperties(QList<P_ID>& dest, bool writeable)
+{
+  DurationElement::supportedProperties(dest, writeable);
+  if (writeable) {
+        dest << P_ID::SMALL << P_ID::BEAM_MODE << P_ID::STAFF_MOVE << P_ID::DURATION_TYPE << P_ID::VISIBLE;
+        } else {
+        dest << P_ID::SMALL << P_ID::BEAM_MODE << P_ID::STAFF_MOVE << P_ID::DURATION_TYPE;
+        }
+}
```
Comment (Jojo-Schmitz) [inline libmscore/chordrest.cpp]: wrong indentation, not only here but in quite a few more places

File: share/plugins/colornotes.qml [hunk 4]
```diff
@@ -12,29 +12,21 @@
...
-               "#009c95", // G
-               "#0071bb", // G#/Ab
-               "#5e50a1", // A
-               "#8d5ba6", // A#/Bb
-               "#cf3e96"  // B
+      property variant colors : [
+               "#e21c48", "#f26622", "#f99d1c",
+               "#ffcc33", "#fff32b", "#bcd85f",
+               "#62bc47", "#009c95", "#0071bb",
+               "#5e50a1", "#8d5ba6", "#cf3e96"
```
Comment (Jojo-Schmitz) [inline share/plugins/colornotes.qml]: Here you're reverting a recent change (of mine)

Comment (rjmatthews62) [reply share/plugins/colornotes.qml]: Yeah, odd that... I'll re-revert.

**Comment by rjmatthews62** [comment]
OK, I have a question to do with the test phase.
Is this a good place to ask? I've never seen any activity on the mailing list..

I found out what was going on, I'd implemented cursor.element as a property and not an invokable.
So I sorted that out to be consistent.
However. in the p1.log.ref, duration is blank, but it is wrong because it wasn't actually returning anything.
It now returns a FractionWrapper (which is actually correct, but should probably have a more sensible toString.

Should I change p1.log.ref to match, or modify duration to return a blank?
[p1.log](https://github.com/musescore/MuseScore/files/1331452/p1.log)

**Comment by rjmatthews62** [comment]
Struggling with rebase and with travis.  Closing until I can get it tidy,

**Comment by lasconic** [comment]
*Should I change p1.log.ref to match, or modify duration to return a blank?*

Do change the reference file so it is correct. The goal of the tests is to catch regressions. Feel free to add more tests if you feel more would be helpful in the future.

**Pull request closed without merging**